### PR TITLE
[user_activity] Use a fixed reference date in the tests

### LINKED ISSUE #2114: Failing tests in `UserActivityTest` due to outdated activity dates in the mockup data
The user activity tests depend on mockup data where the users' last activity dates are fixed. However, the reference date to determine if a user is active or not is dynamic (i.e., "today", meaning the day one of the test runs). Thus, with time passing, all active users became inactive.

```
======================================================================
FAIL: test_check_users (tests.test_user_activity.UserActivityTest)
Test the check_users method
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/smujahid/repos/mozilla/bugbot/venv/lib/python3.10/site-packages/responses/__init__.py", line 210, in wrapper
    return func(*args, **kwargs)
  File "/Users/smujahid/repos/mozilla/bugbot/tests/test_user_activity.py", line 46, in test_check_users
    self.assertEqual(inactive_users.keys(), self.disabled_users)
AssertionError: dict_keys(['mcastelluccio@mozilla.com', '[86 chars]ld']) != {'u709243@disabled.tld', 'balbeza@mozilla[22 chars]com'}

======================================================================
FAIL: test_check_users_keep_active (tests.test_user_activity.UserActivityTest)
Test the check_users method with keep_active set to True
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/smujahid/repos/mozilla/bugbot/venv/lib/python3.10/site-packages/responses/__init__.py", line 210, in wrapper
    return func(*args, **kwargs)
  File "/Users/smujahid/repos/mozilla/bugbot/tests/test_user_activity.py", line 62, in test_check_users_keep_active
    self.assertEqual(user["status"], UserStatus.ACTIVE)
AssertionError: <UserStatus.ABSENT: 5> != <UserStatus.ACTIVE: 1>

======================================================================
FAIL: test_get_bz_users_with_status (tests.test_user_activity.UserActivityTest)
Test the get_bz_users_with_status method
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/smujahid/repos/mozilla/bugbot/venv/lib/python3.10/site-packages/responses/__init__.py", line 210, in wrapper
    return func(*args, **kwargs)
  File "/Users/smujahid/repos/mozilla/bugbot/tests/test_user_activity.py", line 116, in test_get_bz_users_with_status
    self.assertEqual(user["status"], UserStatus.ACTIVE)
AssertionError: <UserStatus.ABSENT: 5> != <UserStatus.ACTIVE: 1>

----------------------------------------------------------------------
Ran 5 tests in 0.070s

FAILED (failures=3)
```

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Fixes #2114

## Checklist

- [x] Type annotations added to new functions
- [x] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/bugbot/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-coverage==7.2.7
+coverage==6.5.0
 mock==5.0.2
-responses==0.23.1
+responses==0.21.0
```
Comment (marco-c) [inline requirements-dev.txt:3]: Why these changes?

Comment (suhaibmujahid) [reply requirements-dev.txt]: Sorry, it was committed by mistake. Thank you for catching it. Reverted in 8acc40aef7dd326b0f5fd056aecf9babdfed36ca.

**Review by marco-c — APPROVED** [review]
